pandas/tests/plotting/test_hist_method.py

```diff
@@ -360,10 +364,16 @@ def test_hist_column_order_unchanged(self, column, expected):
             index=["pig", "rabbit", "duck", "chicken", "horse"],
         )
 
-        axes = _check_plot_works(df.hist, column=column, layout=(1, 3))
-        result = [axes[0, i].get_title() for i in range(3)]
-
-        assert result == expected
+        with tm.assert_produces_warning(UserWarning):
+            # _check_plot_works creates subplots inside,
+            # meanwhile df.hist here creates subplots itself for each column.
+            # It leads to warnings like this:
+            # UserWarning: To output multiple subplots,
+            # the figure containing the passed axes is being cleared
+            # Similar warnings were observed in GH #13188
+            axes = _check_plot_works(df.hist, column=column, layout=(1, 3))
```

Comment:
emm, will `_check_single_plot_works` not raise warning here? seems the purpose of having `_check_single_plot_works` is to avoid creating subplots inside function?

would you clarify when to use `_check_plot_works` and when to choose `_check_single_plot_works` in general plotting tests?

Reply:
Right, ``_check_single_plot_works`` does not raise the warning concerned, regarding figure being cleared.
In my opinion the need for ``_check_single_plot_works`` (or whatever the better name would be) is when the plotting method (like ``DataFrame.hist``) creates subplots itself (may depend on the the options provided to the method).
The warning gets raised when the plotting method creates enough subplots required on its own, but we forcefully add subplots in ``_check_plot_works``.

Reply:
emm, okay. for `DataFrame.hist`, i think as long as it plots data from multiple columns, or a `by` is used, then will have subplots.

my main concern is in `pandas.plot` API where the implementation is different, and there is an argument called `subplots`, then do we also need to use `_check_single_plot_works` (or probably other name is better) when `subplots=True` for those test cases?

Reply:
Instead of putting in-line comments under `_check_plot_works` function, maybe it's nicer to put few words about usecase/guideline of when to suggest to use which so that could help future contributors select the right function? how does this sound?

Reply:
> 
> 
> Instead of putting in-line comments under `_check_plot_works` function, maybe it's nicer to put few words about usecase/guideline of when to suggest to use which so that could help future contributors select the right function? how does this sound?

I actually thought of not using comments here and instead call ``_check_single_plot_works``. This way there will be no warning raised.
```
        axes = _check_single_plot_works(df.hist, column=column, layout=(1, 3))
        result = [axes[0, i].get_title() for i in range(3)]
        assert result == expected
```

Reply:
> 
> 
> emm, okay. for `DataFrame.hist`, i think as long as it plots data from multiple columns, or a `by` is used, then will have subplots.
> 
> my main concern is in `pandas.plot` API where the implementation is different, and there is an argument called `subplots`, then do we also need to use `_check_single_plot_works` (or probably other name is better) when `subplots=True` for those test cases?

Right, is ``subplots=True``, then it seems that we need to use ``_check_single_plot_works``.
But I really don't like the name of this function )

Reply:
> This way there will be no warning raised.

yes, https://github.com/pandas-dev/pandas/pull/37274#discussion_r512218624 actually I was here wondering why not use `_check_single_plot_works` to remove warning

> I actually thought of not using comments

sorry for my confusing comment, i meant to have one sentence to tell future users when to use `_check_single_plot_works` and when to use `_check_plot_works` in testing, something like `Using _check_single_plot_works when plotting methods generate a subplot itself, if not, use _check_plot_works` will be enough IMHO.

Reply:
> But I really don't like the name of this function

indeed, maybe `_check_plot_with_subplots_works`? or something more straightforward/explicit?

Reply:
What about ``_check_plot_default_axes_works``?
I replaced catching warning with the use of the new function as well.